src/decode-icmpv4.c

```diff
@@ -270,6 +271,14 @@ int DecodeICMPV4(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p, const uint8_t
             if (p->icmpv4h->code!=0) {
                 ENGINE_SET_EVENT(p,ICMPV4_UNKNOWN_CODE);
             }
+
+            if (p->payload_len < sizeof(ICMPV4Timestamp)) {
+                ENGINE_SET_EVENT(p, ICMPV4_IPV4_TRUNC_PKT);
+            } else {
+                p->icmpv4vars.hlen += sizeof(ICMPV4Timestamp);
+                p->payload += sizeof(ICMPV4Timestamp);
```

Comment:
I wonder if we can do this in a cleaner way. Maybe set it once correctly, instead of first setting it to a default then adjusting like this.